atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepableCellsReader.java

```diff
@@ -0,0 +1,138 @@
+        return kvs.getRowsColumnRange(TABLE_REF, rows, ALL_COLUMNS, 100_000, SweepQueueUtils.CAS_TS);
+    }
+
+    private void populateResults(SweepableCellsTable.SweepableCellsRow row, Map.Entry<Cell, Value> entry,
+            Map<WriteReference, Long> results) {
+        SweepableCellsTable.SweepableCellsColumn col = SweepableCellsTable.SweepableCellsColumn.BYTES_HYDRATOR
+                .hydrateFromBytes(entry.getKey().getColumnName());
+
+        if (isReferenceToDedicatedRows(col)) {
+            populateFromDedicatedRows(row, col, results);
+        } else {
+            populateFromValue(row, col, entry.getValue(), results);
+        }
+    }
+
+    private boolean isReferenceToDedicatedRows(SweepableCellsTable.SweepableCellsColumn col) {
+        return col.getWriteIndex() < 0;
+    }
+
+    private void populateFromDedicatedRows(SweepableCellsTable.SweepableCellsRow row,
+            SweepableCellsTable.SweepableCellsColumn col, Map<WriteReference, Long> results) {
+        TargetedSweepMetadata metadata = TargetedSweepMetadata.BYTES_HYDRATOR.hydrateFromBytes(row.getMetadata());
+        long partitionFine = row.getTimestampPartition();
+        int numberOfDedicatedRows = (int) -col.getWriteIndex();
+
+        List<byte[]> dedicatedRows = new ArrayList<>();
+        for (int i = 0; i < numberOfDedicatedRows; i++) {
+            byte[] metadataBytes = ImmutableTargetedSweepMetadata.builder()
+                    .from(metadata)
+                    .dedicatedRow(true)
+                    .dedicatedRowNumber(i)
+                    .build()
+                    .persistToBytes();
+            dedicatedRows.add(SweepableCellsTable.SweepableCellsRow.of(partitionFine, metadataBytes).persistToBytes());
+        }
+        RowColumnRangeIterator iterator = getAllColumns(dedicatedRows);
+        iterator.forEachRemaining(entry -> populateResults(row, entry, results));
+    }
+
+    private void populateFromValue(SweepableCellsTable.SweepableCellsRow row,
+            SweepableCellsTable.SweepableCellsColumn col,
+            Value value,
+            Map<WriteReference, Long> results) {
+        WriteReference writeRef = SweepableCellsTable.SweepableCellsColumnValue.hydrateValue(value.getContents());
+
+        long timestamp = row.getTimestampPartition() * SweepQueueUtils.TS_FINE_GRANULARITY + col.getTimestampModulus();
+        addIfMaxForCell(timestamp, writeRef, results);
+    }
+
+    private void addIfMaxForCell(long ts, WriteReference writeRef, Map<WriteReference, Long> result) {
+        result.merge(writeRef, ts, Math::max);
+    }
+
+    private static ColumnRangeSelection allPossibleColumns() {
+        byte[] startCol = SweepableCellsTable.SweepableCellsColumn.of(0L, -TargetedSweepMetadata.MAX_DEDICATED_ROWS)
+                .persistToBytes();
+        byte[] endCol = SweepableCellsTable.SweepableCellsColumn.of(SweepQueueUtils.TS_FINE_GRANULARITY, 0)
+                .persistToBytes();
```

Comment:
yep, confirming this looks like it grabs everything (this is start inclusive end exclusive)